console-csharp-snippets-sample/AuthenticationHelper.cs

```diff
@@ -77,24 +75,17 @@ public static GraphServiceClient GetAuthenticatedClientForUser()
         public static async Task<string> GetTokenForUserAsync()
         {
             AuthenticationResult authResult;
+            IEnumerable<IAccount> accounts = await IdentityClientApp.GetAccountsAsync();
+            IAccount firstAccount = accounts.FirstOrDefault();
             try
             {
-                authResult = await IdentityClientApp.AcquireTokenSilentAsync(Scopes, IdentityClientApp.Users.First());
-                TokenForUser = authResult.AccessToken;
+                authResult = await IdentityClientApp.AcquireTokenSilentAsync(Scopes, firstAccount);
             }
-
-            catch (Exception)
+            catch (MsalUiRequiredException)
             {
-                if (TokenForUser == null || Expiration <= DateTimeOffset.UtcNow.AddMinutes(5))
```

Comment:
Never look at the expiration, and never leverage a stored token. It could have been revoked for other reasons (password changed etc …). This is bad practice!